Sources/Client/Client/Client+Request.swift

```diff
@@ -68,6 +68,36 @@ extension Client {
     func request<T: Decodable>(endpoint: Endpoint,
                                _ progress: @escaping Progress,
                                _ completion: @escaping Completion<T>) -> Cancellable {
+        if let uploader = uploader, let completion = completion as? Completion<FileUploadResponse> {
```

Comment:
Why doesn't `Client` conform to the `Uploader` protocol too? I think it would make the logic simpler and the solution more flexible. 
Or even better! We can extract the logic from `Client` into the `DefaultUploader` which will be used unless you change it to custom one. Then the `Client` wouldn't need to worry at all if it uses the default or a custom uploader.

Reply:
I updated PR and please check my message in Slack.